staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/aes/aes.go

```diff
@@ -48,6 +50,63 @@ func NewGCMTransformer(block cipher.Block) value.Transformer {
 	return &gcm{block: block}
 }
 
+// NewGCMTransformerFromConfig takes a configuration and creates a prefix transformer to
+// handle AEAD encryption for multiple keys. Secrets must be 16, 24 or 32 bytes long,
+// and encoded in base64.
+// Example configuration:
+// keys:
+//   - name: key1
+//     secret: c2VjcmV0IGlzIHNlY3VyZQ==
+//   - name: key2
+//     secret: dGhpcyBpcyBwYXNzd29yZA==
+func NewGCMTransformerFromConfig(config map[string]interface{}) (value.Transformer, error) {
+
+	// Obtain list of keys as []interface{}
+	if keysInterface, ok := config["keys"].([]interface{}); ok {
```

Comment:
We generally don't do this - we define API types and include the config that way.  Anything that reads from an API config should be outside this package (consuming it) and result in a properly instantiated prefix transformer.